metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchQueryBuilder.java

```diff
@@ -72,22 +81,59 @@ private static QueryBuilder buildInternalQuery(@Nonnull EntitySpec entitySpec, @
     return finalQuery;
   }
 
-  private static FunctionScoreQueryBuilder.FilterFunctionBuilder[] buildFilterFunctions(
-      @Nonnull EntitySpec entitySpec) {
-    return entitySpec.getSearchableFieldSpecs()
+  private static FunctionScoreQueryBuilder.FilterFunctionBuilder[] buildScoreFunctions(@Nonnull EntitySpec entitySpec) {
+    List<FunctionScoreQueryBuilder.FilterFunctionBuilder> finalScoreFunctions = new ArrayList<>();
+    // Add a default weight of 1.0 to make sure the score function is larger than 1
+    finalScoreFunctions.add(
+        new FunctionScoreQueryBuilder.FilterFunctionBuilder(ScoreFunctionBuilders.weightFactorFunction(1.0f)));
+    entitySpec.getSearchableFieldSpecs()
         .stream()
         .flatMap(fieldSpec -> fieldSpec.getSearchableAnnotation()
             .getWeightsPerFieldValue()
             .entrySet()
             .stream()
-            .map(entry -> buildFilterFunction(fieldSpec.getSearchableAnnotation().getFieldName(), entry.getKey(),
+            .map(entry -> buildWeightFactorFunction(fieldSpec.getSearchableAnnotation().getFieldName(), entry.getKey(),
                 entry.getValue())))
-        .toArray(FunctionScoreQueryBuilder.FilterFunctionBuilder[]::new);
+        .forEach(finalScoreFunctions::add);
+
+    entitySpec.getSearchScoreFieldSpecs()
+        .stream()
+        .map(fieldSpec -> buildScoreFunctionFromSearchScoreAnnotation(fieldSpec.getSearchScoreAnnotation()))
+        .forEach(finalScoreFunctions::add);
+
+    return finalScoreFunctions.toArray(new FunctionScoreQueryBuilder.FilterFunctionBuilder[0]);
   }
 
-  private static FunctionScoreQueryBuilder.FilterFunctionBuilder buildFilterFunction(@Nonnull String fieldName,
+  private static FunctionScoreQueryBuilder.FilterFunctionBuilder buildWeightFactorFunction(@Nonnull String fieldName,
       @Nonnull Object fieldValue, double weight) {
     return new FunctionScoreQueryBuilder.FilterFunctionBuilder(QueryBuilders.termQuery(fieldName, fieldValue),
         ScoreFunctionBuilders.weightFactorFunction((float) weight));
   }
+
+  private static FunctionScoreQueryBuilder.FilterFunctionBuilder buildScoreFunctionFromSearchScoreAnnotation(
+      @Nonnull SearchScoreAnnotation annotation) {
+    FieldValueFactorFunctionBuilder scoreFunction =
+        ScoreFunctionBuilders.fieldValueFactorFunction(annotation.getFieldName());
```

Comment:
what is a factor function...?

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-field-value-factor